### ISSUE TITLE
[QUESTION] Azure SDK to query Azure HCI

### ISSUE THREAD

**Issue by YeskaNova** [Azure/azure-sdk-for-net#33161]
### Library name and version

Microsoft.Azure.Management.Websites

### Query/Question

Hi,

Our customer doesn't want to use Azure Public Cloud, but he has Azure HCI. We have already some automation tools that target the Public Azure Cloud using the .net Azure SDK libraries. Are these compatible with Azure HCI ( meaning, we only need to change the endpoint to the Azure HCI and everything will work just the same ), or Azure HCI has its own set of management libraries ?

Regards,

Yeska

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.  Please be aware that responses will be delayed due to the US holidays.

**Comment by xboxeer** [comment]
You can try this code
An ArmEnvironment object can let you customize the Azure cloud you want your application to run

```csharp
              var options=new ArmClientOptions();
               var env = new ArmEnvironment(new Uri("your HCI endpoint"), "your audience");
               options. Environment=env;
               var client = new ArmClient(new DefaultAzureCredential(
                   new DefaultAzureCredentialOptions() 
                   { VisualStudioTenantId=context.Configuration.GetValue<string>("VisualStudioTenantId") }),
                   context.Configuration.GetValue<string>("DefaultSubscriptionId"),
                   options);
```

**Comment by YeskaNova** [comment]
Thank you @xboxeer, that's very helpful. Do you know if there is a list of Azure HCI compatibles services ? I couldn't find that anywhere. In our project we need these ones:

![image](https://user-images.githubusercontent.com/4983925/209299030-e1e52b10-cad2-479a-99d2-2ce5c403539f.png)

**Comment by xboxeer** [comment]
@YeskaNova I don't think we have this compatible services list, azure sdk is usually based on latest version of azure rest API, but you can manual overwrite the api versions that sdk is relying on
```csharp
options.SetApiVersion(new Azure.Core.ResourceType("resourceType"), "apiVersion");
```
different api version may have breaking changes, so it is recommend that you test your code before deploy to prod
